packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -329,30 +382,61 @@ export class QueuedRequestController extends BaseController<
     });
   }
 
-  async #waitForDequeue({
-    origin,
-    networkClientId,
-  }: {
-    origin: string;
-    networkClientId: NetworkClientId;
-  }): Promise<void> {
-    const { promise, reject, resolve } = createDeferredPromise({
+  /**
+   * Adds a request to the queue to be processed. A promise is returned that resolves/rejects when
+   * this request should continue execution/fail early. Additionally it returns a callback that
+   * must be called after the request finishes execution.
+   *
+   * Internally, the controller triggers the above returned promise to resolve via the `processRequest`.
+   * It also may await a promise that resolves with the above returned callback.
```

Comment:
Nit: It's not clear to me what this sentence means 🤔 

The rest of the description looks pretty good. Maybe we can just delete this part?

```suggestion
```